extension-compose/src/main/java/com/mapbox/maps/extension/compose/MapboxMap.kt

```diff
@@ -47,4 +101,65 @@ public fun MapboxMap(
     },
     modifier = modifier,
   )
+
+  val parentComposition = rememberCompositionContext()
+  val currentContent by rememberUpdatedState(content)
+
+  LaunchedEffect(Unit) {
+    val composition = Composition(
+      MapApplier(mapView), parentComposition
+    ).apply {
+      setContent {
+        ComposeNode<MapboxMapNode, MapApplier>(
+          factory = {
+            MapboxMapNode(
+              mapView,
+              onMapClickListener,
+              onMapLongClickListener
+            )
+          },
+          update = {
+            // input arguments updater
+            update(onMapClickListener) { listener ->
+              val previousListener = this.clickListener
+              controller.gestures.apply {
+                previousListener?.let {
+                  removeOnMapClickListener(it)
+                }
+                listener?.let {
+                  addOnMapClickListener(it)
+                }
+              }
+              this.clickListener = listener
+            }
+            update(onMapLongClickListener) { listener ->
+              val previousListener = this.longClickListener
+              controller.gestures.apply {
+                previousListener?.let {
+                  removeOnMapLongClickListener(it)
+                }
+                listener?.let {
+                  addOnMapLongClickListener(it)
+                }
+              }
+              this.longClickListener = listener
+            }
+            update(mapInitOptions) {
+              throw IllegalStateException(
+                """
+                Mutating MapInitOptions during composition is not allowed.
+                """.trimIndent()
+              )
+            }
+          }
```

Comment:
@yunikkk @jush continuing the discussion here, for now it just throws when the `MapInitOption` is updated.

Maybe we can also explore other ways to configure it, but allowing to recreate the MapView is too heavy and requires more architecture changes, wdyt?

Reply:
Agree, lets iterate, should be fine for now